addons/mail/static/src/core/common/partner_suggestion_compare_fn.js

```diff
@@ -0,0 +1,141 @@
+/* @odoo-module */
+
+import { cleanTerm } from "@mail/utils/common/format";
+import { registry } from "@web/core/registry";
+
+/**
+ * Registry of functions to sort partner suggestions.
+ * The expected value is an object of the following form:
+ *    { sequence: number, fn: (partner1, partner2, { searchTerms: string, thread?: Thread }) => number|undefined }
+ * Note that the compare functions will be call in the context of the
+ * caller (i.e. the SuggestionService).
+ */
+export const partnerSuggestionCompareRegistry = registry.category(
+    "mail.partner_suggestion_compare_fn"
+);
+
+// Ordering:
+// - recent chat partners
+// - internal users
+// - channel members
+// - thread followers
+// - longgest match of name, alphabetically if having same length of match
+// - longgest match of email, alphabetically if having same length of match
+// - id
```

Comment:
Probably not necessary as it will require extra maintenance, just folding and reading names in registry should be enough?